src/openrct2/paint/tile_element/Paint.Surface.cpp

```diff
@@ -902,6 +902,36 @@ static void viewport_surface_draw_water_side_top(
     viewport_surface_draw_tile_side_top(session, edge, height, terrain, self, neighbour, true);
 }
 
+static std::pair<int32_t, int32_t> surface_get_height_above_water(
+    const SurfaceElement& surfaceElement, const int32_t height, const int32_t surfaceShape)
+{
+    int32_t local_surfaceShape = surfaceShape;
+    int32_t local_height = height;
+
+    if (surfaceElement.GetWaterHeight() > 0)
+    {
+        int32_t waterHeight = surfaceElement.GetWaterHeight();
+        if (waterHeight > height)
+        {
+            local_height += (2 * COORDS_Z_STEP);
+
+            if (waterHeight != local_height || !(local_surfaceShape & TILE_ELEMENT_SURFACE_DIAGONAL_FLAG))
+            {
+                local_height = waterHeight;
+                local_surfaceShape = 0;
+            }
+            else
+            {
+                int32_t bl = (surfaceShape ^ 0xF) << 2;
+                int32_t bh = bl >> 4;
```

Comment:
This is just a 4 bit rotate should use the rotate macros.

Reply:
So if I'm reading this correctly, I'd have to replace the entire else clause with
```
local_surfaceShape = ror4(surfaceShape ^ 0xF, 2);
```
?

Reply:
yeah that would be the same I believe.

Reply:
https://github.com/OpenRCT2/OpenRCT2/pull/10853